### PR TITLE
feat(1577): feat filter events for no builds [2]

### PR DESCRIPTION
**Pull request by esuc4se**
## Context
Add a setting to the pipeline preference to hide many empty events, such as skip ci, or events that have no jobs to start from the event list. (the latest events are not hidden)

Also, fixed a bug that the setting is not reflected immediately when changing the setting of Show Triggers of pipeline preference.

data-schema PR is [here](https://github.com/screwdriver-cd/data-schema/pull/482).

## Objective
Added "Filter events for no builds" to pipeline preference:
スクリーンショット 2022-03-02 20 01 07

filterEventsForNoBuilds OFF:
スクリーンショット 2022-03-02 20 04 25

filterEventsForNoBuilds ON:
スクリーンショット 2022-03-02 20 04 09

## References
- https://github.com/screwdriver-cd/screwdriver/issues/1577
  - adding a toggle switch on a pipeline or an option page to filter empty events.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by esuc4se** [comment]
@adong
It was difficult to implement integration tests for pipeline preference.
(mock of updateFilterEventsForNoBuilds in controller)

Could you please tell me how to write this test?

The tests to add are:
1. The implemented pipeline preference is displayed normally.
2. The appropriate parameters are sent to the API when the toggle switch is turned on.
3. The parameters returned from Mock are normally reflected in the pipeline preference.

File: app/components/pipeline-options/template.hbs [hunk 1]
```diff
@@ -225,6 +225,23 @@
             </div>
           </div>
         </li>
+        <li>
+          <div class="row">
+            <div class="col-xs-10">
+              <h4>Filter Events For No Builds</h4>
+              <p>Setup your pipeline preference to filter events for no builds</p>
```
Comment (tkyi) [inline app/components/pipeline-options/template.hbs:232]: I think the description is a little unclear for me, not sure what might be better
maybe this?
```suggestion
              <p>Setup your pipeline preference to not show events with no builds</p>
```

Comment (esuc4se) [reply app/components/pipeline-options/template.hbs:232]: thank you.
Incorporate that fix.

**Review by sugarnaoming — APPROVED** [review]
LGTM if test is added

File: app/components/pipeline-options/template.hbs [hunk 2]
```diff
@@ -225,6 +225,23 @@
...
+            <div class="col-xs-10">
+              <h4>Filter Events For No Builds</h4>
+              <p>Setup your pipeline preference to not show events with no builds</p>
+            </div>
+            <div class="col-xs-2 right" title="Toggle to {{if filterEventsForNoBuilds "filter" "unfilter"}} events for no builds">
+              {{x-toggle
+                size="small"
+                value=filterEventsForNoBuilds
+                onLabel="filterEventsForNoBuilds:true"
+                offLabel="filterEventsForNoBuilds::false"
```
Comment (sagar1312) [inline app/components/pipeline-options/template.hbs:239]:
```suggestion
                showLabels=false
```

Comment (sagar1312) [reply app/components/pipeline-options/template.hbs:239]: We may also need to update the same for `groupedEvents` and `showEventTriggers`

**Review by tkyi — APPROVED** [review]

**Pull request merged**